tests/testsuite/cache_messages.rs

```diff
@@ -160,7 +159,17 @@ fn clears_cache_after_fix() {
     // Make sure the cache is invalidated when there is no output.
     let p = project().file("src/lib.rs", "fn asdf() {}").build();
     // Fill the cache.
-    p.cargo("check").with_stderr_contains("[..]asdf[..]").run();
+    p.cargo("check")
+        .with_stderr_data(
```

Comment:
While rustc invocation is omitted, I believe we can still wrap it with `str![]`.

Reply:
Thanks, it has been modified.

BTW: Sorry, I'm a bit confused now, what are the criteria for using or not using `str![]`?

Reply:
See #14039 "4. Resolve non-literal deprecations".

To me, I try porting all of the with `str![]` unless it is too specific, fragile, or hard to model with `str![]`.